### PR TITLE
[sonic-platform-modules-dell] Graceful platform reboot

### PR DESCRIPTION
**Pull request by vharish02**
The following commit addresses the graceful unmounting of file
system and graceful shutdown of dockers before calling a
cold reboot which will cause a power cycle of SSD. This ensures
orderly shutdown and no potential corruption of files systems because
of the power cycle to SSD.
This commit will use the existing systemd-reboot service scripts
and override the configuration to do cold reboot for S6100 and
Z9100.
Unit tested the fix and graceful shutdown of file system and
dockers are done with cold reboot.

Signed-off-by: Harish Venkatraman <harish_venkatraman@dell.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-dell/z9100/scripts/override.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[Service]
+ExecStart=
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-dell/z9100/scripts/override.conf]: why do we have a blank line here?

Comment (vharish02) [reply platform/broadcom/sonic-platform-modules-dell/z9100/scripts/override.conf]: ExecStart needs to be cleared before being reassigned

**Comment by lguohan** [comment]
can you provide the test log? since the request is also to 201803 branch, can you provide test log on that branch as well?

**Comment by vharish02** [comment]
> can you provide the test log? since the request is also to 201803 branch, can you provide test log on that branch as well?

I will provide the test cases and test logs for 201803

**Comment by vharish02** [comment]
Attaching test cases and unit test log.

[PlatformReboot-UnitTestLog.docx](https://github.com/Azure/sonic-buildimage/files/3235171/PlatformReboot-UnitTestLog.docx)

**Comment by lguohan** [comment]
can you resolve the conflict?

**Comment by vharish02** [comment]
> can you resolve the conflict?

done.

**Comment by lguohan** [comment]
retest this please

**Pull request merged**